### DISCUSSION TITLE
Is it convenient to provide static library SDK in release version?

### DISCUSSION THREAD

**Discussion by ccworld1000** [rizinorg/rizin#2478] [General]
**Is your feature request related to a problem? Please describe.**
Is it convenient to provide static library SDK in release version?

**Describe the solution you'd like**
It is possible to integrate SDK directly and do some other things

**Describe alternatives you've considered**
It is possible to integrate SDK directly and do some other things

**Comment by ccworld1000** [comment]
Under the macOS platform, I want to try the RZ library and do something.

**Reply by ret2libc** [reply]
We provide static libraries/rizin for linux.
Is that what you are looking for, but for Mac?

**Reply by ccworld1000** [reply]
for macos

**Reply by ccworld1000** [reply]
macho

**Reply by ccworld1000** [reply]
mac env: x86-64

**Comment by ccworld1000** [comment]
There seems to be a misunderstanding about the naming of library files "mach0"

**Comment by ccworld1000** [comment]
For example, most of the functions of my private tool "CCTrackingLibrary" have been implemented. Calling the next version of SDK leads to most incompatibilities, which is equivalent to rewriting again. Too many incompatibilities are equivalent to writing twice. Therefore, I want to try RZ. I hope it is easy to use, stable API and easy to integrate. Part of it is described in RZ document. See if it can replace the previous SDK.

**Reply by ret2libc** [reply]
Sorry I'm not sure I understood you.

**Reply by ccworld1000** [reply]
1. It is expected that the release version contains the binary library file librz(static library file) under the macOS platform

**Reply by ccworld1000** [reply]
2."macho" association naming ambiguity
CC # ls rizin-0.3.4/librz/bin/format/mach0
coresymbolication.c		mach0.h
coresymbolication.h		mach064.c
dyldcache.c			mach064.h
dyldcache.h			mach064_is_kernelcache.c
dyldcache_rebase.c		mach064_rebase.c
fatmach0.c			mach0_defines.h
fatmach0.h			mach0_rebase.c
mach0.c				mach0_specs.h
CC #

**Reply by ccworld1000** [reply]
o not zero
